internal/pipeline/docker/docker.go

```diff
@@ -209,6 +210,9 @@ func dockerTag(ctx *context.Context, image, tag string) error {
 }
 
 func dockerPush(ctx *context.Context, docker config.Docker, image string) error {
+	if err := dockerLogin(ctx, docker); err != nil {
```

Comment:
I'm not sure goreleaser should handle docker logins...

Reply:
@caarlos0 can you explain why?

Reply:
It seems like we need this just because docker in docker, if that's the case, we should probably handle that in the entrypoint script, not on goreleaser itself.

I won't expect that running `goreleaser` would do a `docker login`, I think most people will assume the same...